### PR TITLE
Changed Request message to Shunt and added Trigger and Emit debug helper...

### PR DESCRIPTION
**Pull request by verbocity**
Please review

### CODE REVIEW

File: runtime-api/src/main/scala/io/coral/actors/CoralActor.scala [hunk 1]
```diff
@@ -218,9 +223,13 @@ trait CoralActor extends Actor with ActorLogging {
 	def state: Map[String, JValue]
 
 	def stateReceive: Receive = {
-		case GetField(x) => {
+		case GetField(x) =>
 			val value = state.get(x)
-			sender ! render(state)
-		}
+
```
Comment (natbusa) [inline runtime-api/src/main/scala/io/coral/actors/CoralActor.scala:228]: Verbose.

Comment (verbocity) [inline runtime-api/src/main/scala/io/coral/actors/CoralActor.scala:228]: What do you mean? It is your own code...

File: runtime-api/src/main/scala/io/coral/actors/Messages.scala [hunk 2]
```diff
@@ -4,26 +4,34 @@ import akka.actor.{ActorRef, ActorPath}
...
+    // RuntimeActor messages
+    case class CreateActor(json: JObject)
+    case class RegisterActorPath(id: Long, path: ActorPath)
+    case class GetCount()
+    case class ListActors()
+    case class GetActorPath(id: Long)
+    case class Delete(id: Long)
+    case class DeleteAllActors()
+
+    // CoralActor messages
```
Comment (natbusa) [inline runtime-api/src/main/scala/io/coral/actors/Messages.scala:16]: What does it mean, if an actor passes another actor a command Emit?
Why is this semantic being introduced? What is the use case?

File: runtime-api/src/main/scala/io/coral/actors/CoralActor.scala [hunk 3]
```diff
@@ -177,35 +177,40 @@ trait CoralActor extends Actor with ActorLogging {
...
+                    transmit(result)
+                    if (result != JNothing) s ! result
+
+                case None => log.warning("some variables are not available")
+            }
+
+            r.onFailure {
+                case e => println(e)
+            }
+        case Trigger(json) =>
```
Comment (natbusa) [inline runtime-api/src/main/scala/io/coral/actors/CoralActor.scala:209]: Trigger does not check exceptions. Why?

File: runtime-api/src/main/scala/io/coral/actors/Messages.scala (hunk 2 above)
Comment (verbocity) [inline runtime-api/src/main/scala/io/coral/actors/Messages.scala:16]: The use case is testing. It is useful to execute the emit function specifically, and nothing else, from a unit test. This way, it is very easy to intercept and check an emit result.

File: runtime-api/src/main/scala/io/coral/actors/CoralActor.scala (hunk 3 above)
Comment (verbocity) [inline runtime-api/src/main/scala/io/coral/actors/CoralActor.scala:209]: We could add more detailed exception handling but for now, this was also not present in the original trigger call, only if the asynchronous call itself has succeeded or not.

**Comment by natbusa** [comment]
It seems that the coral semantic is modified to fit some specific use case of the cassandraActor. Why?
What are those use cases? The current semantic of the Coral actor is to process data. This pull request introduces a new semantic of actors passing each others commands, in some case without data. 

Currently coral actors are passing each other data, hence they build up a dataflow. These changes transform the data flow in a collection of RPC objects. This is a very different paradigm, and I would be strongly discourage, since it stimulated the use of actors as remote procedure agents.

**Comment by verbocity** [comment]
The CoralActor semantics are not modified, the existing semantics are left intact. 
The Trigger and Emit messages are introduced to facilitate testing of the trigger and emit functions. The only case where you ever want to use the Trigger and Emit messages is in test classes.

The emit function always takes a json object as input, but in the current usage, it is almost never used. This is because there are only a few exceptions in which you want to work with the original json in the emit function. 

I do not see how the changes transform any data flow, and I do not understand the connection with RPC objects. There is also no paradigm shift.

**Pull request merged**